pydantic/main.py

```diff
@@ -1473,6 +1474,26 @@ def create_model(  # noqa: C901
                     'Field definitions should be a `(<type>, <default>)`.',
                     code='create-model-field-definitions',
                 ) from e
+
+        elif (
+            sys.version_info >= (3, 9) and f_def.__class__ == typing._AnnotatedAlias  # pyright: ignore[reportGeneralTypeIssues]
+        ):  # typing.Annotated introduced on python3.9 https://typing.readthedocs.io/en/latest/spec/qualifiers.html#annotated
```

Comment:
I think you can make use of the utils in `pydantic._internal._typing_extra`, e.g. `is_annotated`

Reply:
Thanks for the suggestion! Though I'm wondering why we aren't using the `typing._AnnotatedAlias` and instead use a workaround. But seems this is a viable solution.

Reply:
`typing._AnnotatedAlias` is an implementation detail, and `is_annotated` relies on [`get_origin`](https://docs.python.org/3/library/typing.html#typing.get_origin) which is the documented way of getting the unsubscripted version of the `Annotated` special form.

Instead, I would do something like:

```python
        elif  _typing_extra.is_annotated(f_def):
            try:
                f_annotation = (
                    f_def.__origin__
                )  # __origin__ represents the annotation_type in Annotated[annotate_type, ...]

                f_value = f_def.__metadata__[
                    0
                ]  # Python Annotated requires at least one python variable to represent the annotation
            except ValueError as e:
                raise PydanticUserError(
                    'Field definitions should be a `typing.Annotation[type, definition]`.',
                    code='create-model-field-definitions',
                ) from e
```

The Python version check is not necessary, as `is_annotated` will safely return `False` if it is not an instance of `Annotated`. Plus people can make use of `typing_extensions.Annotated` (and `is_annotated` can handle both).

I'll note that accessing `__origin__` [^1] and `__metadata__` is not documented, although I don't think the implementation will change anytime.

One way of doing it would be:

`f_annotation, f_value = typing_extensions.get_args(f_def)`

Considering `Annotated` is guaranteed to have a type and at least one metadata value. In Python 3.10, I get `TypeError: Annotated[...] should be used with at least two arguments (a type and an annotation).` when trying to do `Annotated[int]`, but I'm not sure this is the case on 3.8/3.9.

[^1]: Not to be confused with the return value of `get_origin` (yeah this is weird). `Annotated[int, ""].__origin__` returns `int`, while `get_origin(Annotated[int, ""])` return the `Annotated` class.